src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -1204,6 +1189,24 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                 arg_auto_root = GPT_AUTO_ROOT_OFF;
                 log_debug("Disabling root partition auto-detection, roothash= is set.");
 
+                arg_verity_settings.designator = PARTITION_ROOT;
+
+                r = unhexmem(value, &arg_verity_settings.root_hash, &arg_verity_settings.root_hash_size);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to parse root hash from kernel command line switch: %m");
```

Comment:
Please don't obfuscate the error message by using a worded description. "Failed to parse roothash= from the kernel commandline: %m".